tests/integration/components/select-component-test.js

```diff
@@ -133,3 +138,24 @@ test('It displays the specified component when componentNameForGroupTooltip is p
     'the groupItem argument is passed'
   );
 });
+
+test('It does not display loading text and empty content component at the same time', async function(assert) {
+  this.container.register('template:components/some-component', hbs`<span data-test-some-component>No results</span>`);
+
+  this.set('content', [
+    'foo', 'bar', 'bar', 'baz'
+  ]);
+
+  await this.render(hbs`
+  {{select-component
+      content=content
+      emptyContentView='some-component'
+  }}`);
+
+  await this.helpers.inputSearchText('no match');
+
+  let loadingIsVisible = this.$('.ember-select-loading').length;
+  let noResultsIsVisible = this.$('.ember-select-empty-content').length;
+
```

Comment:
make sure that your test fails before your change, ideally in your test, `isLoading` and `hasNoResults` should both true and then make sure that the loading message shows up but not no results